src/qt/bitcoingui.h

```diff
@@ -161,6 +161,7 @@ class BitcoinGUI : public QMainWindow
     RPCConsole* rpcConsole = nullptr;
     HelpMessageDialog* helpMessageDialog = nullptr;
     ModalOverlay* modalOverlay = nullptr;
+    bool closeEventRegistered = false;
```

Comment:
This is used to prevent calling `qApp->quit()` more than once. Is this a real problem - could this be removed?

Reply:
That's right. This is needed since the reordering in fd6d499. The second call to ```aApp-quit()``` would make the QEventLoop exit before the RPC executor thread finishes causing the application to crash.
I've made sure that closing bitcoin QT by typing in 'stop' in the console still works as intended when this line was introduced in 9a2305a